src/main/scala/firrtl/annotations/MemoryInitAnnotation.scala

```diff
@@ -25,20 +26,29 @@ case class MemoryRandomInitAnnotation(target: ReferenceTarget) extends MemoryIni
   override def duplicate(n: ReferenceTarget): Annotation = copy(n)
   override def initValue:    MemoryInitValue = MemoryRandomInit
   override def isRandomInit: Boolean = true
+  override def dedup: Option[((ReferenceTarget, Any), Annotation, ReferenceTarget)] = Some(
+    ((Utils.localizeTarget(target), Nil), copy(target = Utils.localizeTarget(target)), target)
+  )
 }
 
 /** Initialize all entries of the `target` memory with the scalar `value`. */
 case class MemoryScalarInitAnnotation(target: ReferenceTarget, value: BigInt) extends MemoryInitAnnotation {
   override def duplicate(n: ReferenceTarget): Annotation = copy(n)
   override def initValue:    MemoryInitValue = MemoryScalarInit(value)
   override def isRandomInit: Boolean = false
+  override def dedup: Option[((ReferenceTarget, Any), Annotation, ReferenceTarget)] = Some(
+    ((Utils.localizeTarget(target), value), copy(target = Utils.localizeTarget(target)), target)
+  )
```

Comment:
I think these overrides need to keep it private:
```suggestion
  override private[firrtl] def dedup: Option[((ReferenceTarget, Any), Annotation, ReferenceTarget)] = Some(
    ((Utils.localizeTarget(target), value), copy(target = Utils.localizeTarget(target)), target)
  )
```